app/src/org/commcare/android/view/CustomButtonWithText.java

```diff
@@ -0,0 +1,126 @@
+package org.commcare.android.view;
+
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.Color;
+import android.graphics.drawable.ColorDrawable;
+import android.graphics.drawable.Drawable;
+import android.graphics.drawable.StateListDrawable;
+import android.os.Build;
+import android.util.AttributeSet;
+import android.util.StateSet;
+import android.widget.ImageButton;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import org.commcare.dalvik.R;
+
+/**
+ * Custom button that allows setting image, text, and color
+ *
+ * @author Daniel Luna (dluna@dimagi.com)
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public abstract class CustomButtonWithText extends RelativeLayout {
+    static final int DEFAULT_TEXT_COLOR = R.color.cc_core_bg;
+    ImageButton button;
+    TextView textView;
+
+    public CustomButtonWithText(Context context) {
+        super(context);
+    }
+
+    public CustomButtonWithText(Context context, AttributeSet attrs) {
+        super(context, attrs);
+
+        inflateAndExtractCustomParams(context, attrs);
+    }
+
+    public CustomButtonWithText(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+
+        inflateAndExtractCustomParams(context, attrs);
+    }
+
+    protected void inflateAndExtractCustomParams(Context context, AttributeSet attrs) {
+        inflate(context, getLayoutResourceId(), this);
+        this.setClickable(true);
+
+        TypedArray typedArray = context.obtainStyledAttributes(attrs, R.styleable.CustomButtonWithText);
+
+        Drawable backgroundImg = typedArray.getDrawable(R.styleable.CustomButtonWithText_img);
+        int backgroundColor = getResources().getColor(typedArray.getResourceId(R.styleable.CustomButtonWithText_backgroundColor, android.R.color.transparent));
+        String text = typedArray.getString(R.styleable.CustomButtonWithText_subtitle);
+        int colorButtonText = typedArray.getResourceId(R.styleable.CustomButtonWithText_textColor, DEFAULT_TEXT_COLOR);
+
+        typedArray.recycle();
+
+        button = (ImageButton)findViewById(R.id.button);
+        textView = (TextView) findViewById(R.id.text_view);
+
+        if (isInEditMode()) {
+            setUI(R.color.cc_brand_color, getResources().getDrawable(R.drawable.barcode), "Your text goes here", colorButtonText);
+        }
+
+        setUI(backgroundColor, backgroundImg, text, colorButtonText);
+    }
+
+    abstract int getLayoutResourceId();
+
+    protected void setUI(int backgroundColor, Drawable backgroundImg, String text, int colorButtonText) {
+        setColor(backgroundColor);
+        setImage(backgroundImg);
+        setText(text);
+        setTextColor(colorButtonText);
+    }
+
+    public void setText(String text) {
```

Comment:
Oh, good call! This one actually needs to be public because it is used in `SetupKeepInstallFragment`, but a bunch of other fields/methods here needed to be private